log/sequencer.go

```diff
@@ -189,10 +191,22 @@ func (s Sequencer) sequenceLeaves(mt *merkle.CompactMerkleTree, leaves []*trilli
 		}
 		// The leaf has now been sequenced.
 		leaves[i].LeafIndex = seq
-		leaves[i].IntegrateTimestamp, err = ptypes.TimestampProto(s.timeSource.Now())
+		integrateTS := s.timeSource.Now()
+		leaves[i].IntegrateTimestamp, err = ptypes.TimestampProto(integrateTS)
 		if err != nil {
 			return nil, nil, fmt.Errorf("got invalid integrate timestamp: %v", err)
 		}
+
+		// Old leaves might not have a QueueTimestamp, only calculate the merge delay if it does.
+		if leaves[i].QueueTimestamp != nil && leaves[i].QueueTimestamp.Seconds != 0 {
+			queueTS, err := ptypes.Timestamp(leaves[i].QueueTimestamp)
+			if err != nil {
+				return nil, nil, fmt.Errorf("got invalid queue timestamp: %v", queueTS)
```

Comment:
Do you want to log and proceed as if the leaf had a zero QueueTimestamp in this case, so we're a bit more resilient?

(This is a bit hypothetical anyway, so it might make no difference in practice.)

Reply:
Yeah, I went back and forward on this myself...
It's hard to see how this could fail, or if it does, then something is probably corrupt, so maybe it's best to bail and get operator intervention.